src/analyses/loopTermination.ml

```diff
@@ -8,12 +8,13 @@ open TerminationPreprocessing
 let loop_counters : stmt VarToStmt.t ref = ref VarToStmt.empty
 
 (** Checks whether a variable can be bounded. *)
-let check_bounded ctx varinfo =
+let ask_bound ctx varinfo =
   let open IntDomain.IntDomTuple in
   let exp = Lval (Var varinfo, NoOffset) in
   match ctx.ask (EvalInt exp) with
-  | `Top -> false
-  | `Lifted v -> not (is_top_of (ikind v) v)
+  | `Top -> `Top
+  | `Lifted v when is_top_of (ikind v) v -> `Top
+  | `Lifted v -> `Lifted v
   | `Bot -> failwith "Loop counter variable is Bot."
```

Comment:
It looks like this could be moved inside the `special`?

Reply:
It could be, but `special` already has very deep matching, which wouldn't make it nicer. Having it at the top level is like `ThreadEscape.has_escaped`, although currently no other analysis needs it.